homeassistant/components/group.py

```diff
@@ -112,6 +113,11 @@ def setup(hass, config):
 
 class Group(object):
```

Comment:
I can definitely make a Visibility Manager ABC that components must inherit to support visibility. This could have the hidden property getter and setter as well as the instance variable visibility that hold the data from the config file.

After that, however, it would have to be the responsibility of the component to add the hidden property to its attributes dictionary. It would work, but I fear in may cause some confusion down the road where people think that since they inherited the ABC, visibility should just work.

I just had a thought, though, all of this could be avoided if visibility is managed at the state level and not the entity level. So the entity could make its suggestion in the attributes dictionary that it uses to create a state, then the state will check against the global configuration. This would make all entities support the visibility property with no additional code and reduce confusion. The down side, however, would be that I don't know if it is a great practice to have the state class manipulating the attributes they are given. That could also lead to confusion. 

What do you think? I think they are both equally good options with their own merits.